src/main/scala/akka/io/AsyncDnsResolver.scala

```diff
@@ -9,32 +11,35 @@ import ru.smslv.akka.dns.raw.{AAAARecord, ARecord, Answer, CNAMERecord, DnsClien
 
 import scala.annotation.tailrec
 import scala.collection.JavaConversions._
-import scala.collection.{breakOut, immutable, mutable}
+import scala.collection.{breakOut, immutable}
+import scala.concurrent.duration.{Deadline, FiniteDuration}
+import scala.io.Source
 import scala.util.Random
-
-private case class CurrentRequest(client: ActorRef,
-                                  name: String,
-                                  ipv4: Option[immutable.Seq[Inet4Address]],
-                                  ipv6: Option[immutable.Seq[Inet6Address]],
-                                  srv: Option[immutable.Seq[SRVRecord]],
-                                  ttl: Long)
+import scala.util.control.NonFatal
 
 class AsyncDnsResolver(cache: SimpleDnsCache, config: Config) extends Actor with ActorLogging {
 
   import AsyncDnsResolver._
 
-  private val nameServers: immutable.Seq[InetSocketAddress] = config.getStringList("nameservers").map(parseNameserverAddress)(breakOut)
+  private val systemNameServers =
+    if (config.getBoolean("resolv-conf"))
+      parseSystemNameServers(Paths.get("/etc/resolv.conf").toFile)
+    else
+      Option.empty[immutable.Seq[InetSocketAddress]]
+  private val nameServers: immutable.Seq[InetSocketAddress] =
+    systemNameServers.getOrElse(config.getStringList("nameservers").map(parseNameserverAddress)(breakOut))
   private val resolveIpv4 = config.getBoolean("resolve-ipv4")
   private val resolveIpv6 = config.getBoolean("resolve-ipv6")
   private val resolveSrv = config.getBoolean("resolve-srv")
   private val negativeTtl = config.getDuration("negative-ttl", TimeUnit.MILLISECONDS)
   private val minPositiveTtl = config.getDuration("max-positive-ttl", TimeUnit.MILLISECONDS)
   private val maxPositiveTtl = config.getDuration("max-positive-ttl", TimeUnit.MILLISECONDS)
+  private val requestTtl = FiniteDuration(config.getDuration("request-ttl", TimeUnit.MILLISECONDS), TimeUnit.MILLISECONDS)
 
   private val random = new Random()
 
   private var requestId: Short = 0
-  private val requests = mutable.OpenHashMap[Short, CurrentRequest]()
+  private var requests = List.empty[(Short, CurrentRequest)]
```

Comment:
You mean a Seq vs a Map? It is a trade off. We must still scan to remove expired entries - a very real issue given that there witnessed memory retention issues. I'm also careful re the position of elements in the sequence. Oldest requests are first with replies shifted to the front.